swarm/api/http/server_test.go

```diff
@@ -892,3 +894,268 @@ func TestMethodsNotAllowed(t *testing.T) {
 	}
 
 }
+
+// HTTP convenience function
+func httpDo(httpMethod string, url string, reqBody io.Reader, headers map[string]string, verbose bool, t *testing.T) (*http.Response, string) {
+	// Build the Request
+	req, err := http.NewRequest(httpMethod, url, reqBody)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+	if verbose {
+		t.Log(req.Method, req.URL, req.Header, req.Body)
+	}
+
+	// Send Request out
+	httpClient := &http.Client{}
+	res, err := httpClient.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Read the HTTP Body
+	buffer, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer res.Body.Close()
+	body := string(buffer)
+
+	return res, body
+}
+
+func TestGet(t *testing.T) {
+	// Setup Swarm
+	srv := testutil.NewTestSwarmServer(t, serverFunc)
+	defer srv.Close()
+	
+	type GetTestCase struct{
```

Comment:
you don't have to declare it as a type, it can all be inlined as an anonymous struct in the `for ... range` expression